packages/fxa-content-server/app/scripts/views/settings/subscription.js

```diff
@@ -30,22 +30,13 @@ const View = FormView.extend({
   },
 
   afterRender() {
-    return this._hasActiveSubscriptions().then(hasSubs => {
-      if (hasSubs) {
-        this.$el.html(this.renderTemplate(Template));
-      }
-    });
-  },
-
-  /**
-   * Check to see if the account has any active subscriptions.
-   *
-   * @returns {Promise} resolves to a bool.
-   */
-  _hasActiveSubscriptions() {
     return this.getSignedInAccount()
-      .fetchActiveSubscriptions()
-      .then(subscriptions => subscriptions.length > 0);
+      .hasActiveSubscriptions()
+      .then(hasSubs => {
+        if (hasSubs) {
+          this.$el.html(this.renderTemplate(Template));
```

Comment:
If there are no subscriptions, does this panel appear empty? What you can do is instead of calling render here and having template() return an empty string is use the fact that if `afterRender` returns false, the panel shouldn't render:

```js

template: Template,
...
afterRender() {
  return this.getSignedInAccount()
  .hasActiveSubscriptions()
  .then(hasSubs => {
    // view is not rendered if hasSubs is falsey.
    return !! hasSubs;
  });
}
```

I think that should work within settings panels too, it definitely works in normal views.